### PR TITLE
Use correct scissor rects

### PR DESCRIPTION
**Pull request by roamic**
Fixes lightning issues in DS3 and BB.

Thanks @tGecko for finding the root cause.

### CODE REVIEW

**Comment by tGecko** [comment]
**oh... my... god**

image


image

**Comment by roamic** [comment]
These changes are breaking some games. This PR will remain a draft until the proper solution is found.

**Comment by rafael-57** [comment]
image
image

[shad_log - Copy.txt](https://github.com/user-attachments/files/17178651/shad_log.-.Copy.txt)

Doesn't only fix lightning intensity but fixes some SFX too! 

The performance improvement is crazy. On my RTX 3080 performance went from 24~29 fps on 1080p to locked 60fps. Before I needed to play at 720p for 60fps.

**Comment by bigol83** [comment]
Red dead redemption hangs almost immediately with this PR 

This is the log sync + Vulkan validations 
[shad_log.txt](https://github.com/user-attachments/files/17178678/shad_log.txt)

**Comment by tGecko** [comment]
fixes lighting in DS3 as well
image

**Comment by Deathedit** [comment]
RDR is black screen now, but game is running and sounds working  
[shad_log.txt](https://github.com/user-attachments/files/17179287/shad_log.txt)

**Comment by squidbus** [comment]
> RDR is black screen now, but game is running and sounds working

This has already been fixed, you are using an older build of the PR.

**Comment by f1amy** [comment]
bloodborne crashed with out of memory
[shad_log.txt](https://github.com/user-attachments/files/17179304/shad_log.txt)

**Comment by Deathedit** [comment]
> > RDR is black screen now, but game is running and sounds working
> 
> This has already been fixed, you are using an older build of the PR.

yes i did, my bad

File: src/video_core/renderer_vulkan/vk_rasterizer.cpp [hunk 1]
```diff
@@ -342,6 +342,14 @@ void Rasterizer::UpdateDynamicState(const GraphicsPipeline& pipeline) {
     }
 }
 
+static s16 CombinedScissorValueTL(s16 scr, s16 win, s16 gen, s16 win_offset) {
```
Comment (raphaelthegreat) [inline src/video_core/renderer_vulkan/vk_rasterizer.cpp]: These can be lambdas in UpdateViewportScissorState

Comment (roamic) [reply src/video_core/renderer_vulkan/vk_rasterizer.cpp]: Done

**Comment by yoplat** [comment]
Will this be merged? Seems to have no downsides and massively improve emulation in bloodborne

**Comment by pwnedbygary** [comment]
> Will this be merged? Seems to have no downsides and massively improve emulation in bloodborne

Did you not see the previous comment? "These changes are breaking some games. This PR will remain a draft until the proper solution is found."

**Review by polybiusproxy — APPROVED** [review]
Looks good to me

**Comment by rafael-57** [comment]
> > Will this be merged? Seems to have no downsides and massively improve emulation in bloodborne
> 
> Did you not see the previous comment? "These changes are breaking some games. This PR will remain a draft until the proper solution is found."

That's old. These issues have been fixed 2 days ago and the issue is not marked WIP anymore

**Comment by yoplat** [comment]
> > Will this be merged? Seems to have no downsides and massively improve emulation in bloodborne
> 
> Did you not see the previous comment? "These changes are breaking some games. This PR will remain a draft until the proper solution is found."

It seems to me like you are the one that doesn't know how to read

**Comment by pwnedbygary** [comment]
> > > Will this be merged? Seems to have no downsides and massively improve emulation in bloodborne
> > 
> > 
> > Did you not see the previous comment? "These changes are breaking some games. This PR will remain a draft until the proper solution is found."
> 
> That's old. These issues have been fixed 2 days ago and the issue is not marked WIP anymore

Derp, egg on my face XD

**Comment by Djipi71** [comment]
not available to test?

**Pull request merged**